DownFall/Models/CarryModel.swift

```diff
@@ -15,30 +15,47 @@ struct CarryModel: Decodable, Equatable {
         return items.contains { $0.type == .gem }
     }
     
-    var totalGold: Int {
-        return items.filter({ $0.type == .gold }).count
+    private var totalGold: Int {
+        return items.filter({ $0.type == .gold }).first?.amount ?? 0
     }
     
-    func pay(_ price: Int) -> CarryModel {
-        var newItems: [Item] = []
-        var pricePaid = 0
-        for item in items {
-            if item.type == .gold && pricePaid < price {
-                pricePaid += 1
-            } else {
-                newItems.append(item)
-            }
+    private var totalGem: Int {
+        return items.filter({ $0.type == .gem }).first?.amount ?? 0
+    }
+    
+    func total(in currency: Currency) -> Int {
+        switch currency {
+        case .gem:
+            return totalGem
+        case .gold:
+            return totalGold
         }
+    }
+    
+    func pay(_ price: Int, inCurrency currency: Currency) -> CarryModel {
+        let itemType: Item.ItemType = currency == .gold ? .gold : .gem
+        
+        return CarryModel(items: items.map { item in
+                if item.type == itemType {
+                    return Item(type: itemType, amount: item.amount - price)
+                } else {
+                    return item
+                }
+            }
+        )
```

Comment:
Currently, we can pay and go into the red.  This seems like a bug in the making.  Lets add a guard that we canAfford and throw an error that the caller must call canAfford before buying something